README.md

```diff
@@ -8,61 +8,79 @@ Head over to the [marketplace](https://www.contentful.com/marketplace/) and foll
+  - Additionally, ensure that the package version number is below 1.0.0. The version of 1.0.0 is reserved for initial release.
+
+- Have a `LICENSE` file at the root of your app directory.
+
+- Use [TypeScript](https://www.typescriptlang.org/). We require this to ensure code quality and maintainability for the marketplace apps.
+
+- Ensure dependencies are up to date. The validation checks will provide warnings if a new app has outdated dependencies.
+  > [!NOTE] > **Please note that dependabot is enabled in this repo and will run automatically.**
+  > While it will attempt to only upgrade dependencies without breaking changes, it is possible it will still introduce issues. Creating and maintaining comprehensive tests is critical to preventing code and build issues. See [Keeping dependencies up to date](#keeping-dependencies-up-to-date) for more details.
+
+### After a new app is introduced
+
+After a new app has been introduced into the repo, the Contentful ecosystem team will make a couple of updates:
+
+- Add the `deploy` script
+
+  - All code deployed by the `marketplace-partner-apps` repo must be hosted by Contentful. Your deploy job will look something like this:
+    `contentful-app-scripts upload --ci --bundle-dir ./build --organization-id ${DEFINITIONS_ORG_ID} --definition-id APP_DEFINITION_ID --token ${CONTENTFUL_CMA_TOKEN}` - Make sure `--bundle-dir` points to the output of your build process (e.g. `./build`). - `DEFINITIONS_ORG_ID` and `CONTENTFUL_CMA_TOKEN` do not need to be changed and will be appropriately filled by the build process
+  - **The Contentful ecosystem team will update the value of `APP_DEFINITION_ID` to a shared, official space. Please do not update it in subsequent changes/pull requests.**
+
+- Add the new app to the repo's dependabot configuration
+
+- Add the new app to the release-please-config.json file so that the release workflow will include the new app. The release workflow encapsulates versioning as well as the creation and maintenance of a changelog
 
 ### Committing and updating your repo
 
-When you are ready to submit your app to the [marketplace apps repo](https://github.com/contentful/marketplace-partner-apps), clone the repo locally, create a new branch with a descriptive name for the app/changes you are introducing, and create a pull-request against the `main` branch. An ecosystems integrations will review your PR soon and let you know if any changes or clarifications are needed. To ensure the fastest approval time on your PR, make sure you:
+When you are ready to submit your app to the [marketplace apps repo](https://github.com/contentful/marketplace-partner-apps), clone the repo locally, create a new branch with a descriptive name for the app/changes you are introducing, and create a pull-request against the `main` branch. An ecosystem team member will review your PR soon and let you know if any changes or clarifications are needed. To ensure the fastest approval time on your PR, make sure you:
```

Comment:
"create a new branch with a descriptive name" should be? "create a new branch with a name that follows Conventional Commits standards (see below)"